packages/grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -350,18 +356,19 @@ export const KeyboardNavigationMixin = (superClass) =>
           dy = -1;
           break;
         case 'PageDown':
-          dy = visibleItemsCount;
+          {
+            const currentRowIndex = this.__getIndexInGroup(activeRow, this._focusedItemIndex);
+            this.scrollToIndex(currentRowIndex); // scroll the current row to the top...
```

Comment:
The `scrollToIndex` method takes an index array for a hierarchical grid so it doesn't work well when there are some expanded items:

https://github.com/vaadin/web-components/assets/1222264/375026ce-c63d-4af3-9eca-d146982a0dff



Scroll to flat index instead: `this._scrollToFlatIndex(activeRow.index);`